src/commands/registerCommands.ts

```diff
@@ -139,49 +139,129 @@ for (const label of bindingTypes.keys()) {
     }
-    const bindingType = bindingTypes.get(pickedBindingTypeOption.label);
-
-    // Open a dialog to select the XSD, DTD to bind.
-    const options: OpenDialogOptions = {
-      canSelectMany: false,
-      openLabel: 'Select XSD or DTD file',
-      filters: {
-        'Grammar files': ['xsd', 'dtd']
+  };
+
+  const fileUri = await window.showOpenDialog(options);
+  if (fileUri && fileUri[0]) {
+    // The XSD, DTD has been selected, get the proper syntax for binding this grammar file in the XML document.
+    const identifier = TextDocumentIdentifier.create(documentURI.toString());
+    const grammarURI = fileUri[0];
+    try {
+      const result = await commands.executeCommand(ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
+      // Insert the proper syntax for binding
+      const lspTextDocumentEdit = <TextDocumentEdit>result;
+      const workEdits = new WorkspaceEdit();
+      for (const edit of lspTextDocumentEdit.edits) {
+        workEdits.replace(documentURI, languageClient.protocol2CodeConverter.asRange(edit.range), edit.newText);
       }
+      workspace.applyEdit(workEdits); // apply the edits
+
+      // Hide the "Bind to grammar/schema" command
+      commands.executeCommand('setContext', 'canBindGrammar', false)
+
+    } catch (error) {
+      window.showErrorMessage('Error during grammar binding: ' + error.message);
     };
+  }
+}
 
-    const fileUri = await window.showOpenDialog(options);
-    if (fileUri && fileUri[0]) {
-      // The XSD, DTD has been selected, get the proper syntax for binding this grammar file in the XML document.
-      const identifier = TextDocumentIdentifier.create(documentURI.toString());
-      const grammarURI = fileUri[0];
-      try {
-        const result = await commands.executeCommand(ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
-        // Insert the proper syntax for binding
-        const lspTextDocumentEdit = <TextDocumentEdit>result;
-        const workEdits = new WorkspaceEdit();
-        for (const edit of lspTextDocumentEdit.edits) {
-          workEdits.replace(documentURI, languageClient.protocol2CodeConverter.asRange(edit.range), edit.newText);
-        }
-        workspace.applyEdit(workEdits); // apply the edits
-      } catch (error) {
-        window.showErrorMessage('Error during grammar binding: ' + error.message);
-      };
+/**
+ * Register commands used for associating grammar file (XSD,DTD) to a given XML file for command menu and CodeLens
+ *
+ * @param context the extension context
+ * @param languageClient the language server client
+ */
+ function registerAssociationCommands(context: ExtensionContext, languageClient: LanguageClient) {
+  // For CodeLens
+  context.subscriptions.push(commands.registerCommand(ClientCommandConstants.OPEN_BINDING_WIZARD, async (uriString: string) => {
+    grammarAssociationCommand(uriString, languageClient)
```

Comment:
```ts
const documentURI = Uri.parse(uriString);
grammarAssociationCommand(documentURI, languageClient);
```

Reply:
This URIString is passed into the registered command as a parameter, as you originally did when you added this command. This is a wrapper, so I believe if I change this, it will change your implementation of the association command.

Reply:
if you see https://github.com/redhat-developer/vscode-xml/blob/a720559fae8f8dd93c14a152dc3a8aa066e9732b/src/commands/registerCommands.ts#L149 the follwing code use an Uri instance and nor a string. So grammarAssociationCommand can have a Uri instanceisntead of String.

In case with Codelens, we convert uri string to an Uri before calling grammarAssociationCommand . In the case of command palette we have laready Uri instance from textDocument, so grammarAssociationCommand  can have an Uri instance instead of string.